tests/Type/Doctrine/data/QueryResult/queryResult.php

```diff
@@ -143,45 +143,391 @@ public function testReturnTypeOfQueryMethodsWithExplicitObjectHydrationMode(Enti
 	}
 
 	/**
-	 * Test that we properly infer the return type of Query methods with explicit hydration mode that is not HYDRATE_OBJECT
+	 * Test that we properly infer the return type of Query methods with explicit hydration mode of HYDRATE_ARRAY
 	 *
-	 * We are never able to infer the return type here
+	 * We can infer the return type by changing every object by an array
 	 */
-	public function testReturnTypeOfQueryMethodsWithExplicitNonObjectHydrationMode(EntityManagerInterface $em): void
+	public function testReturnTypeOfQueryMethodsWithExplicitArrayHydrationMode(EntityManagerInterface $em): void
 	{
 		$query = $em->createQuery('
 			SELECT		m
 			FROM		QueryResult\Entities\Many m
 		');
 
 		assertType(
-			'mixed',
+			'list<array>',
```

Comment:
We should be more specific here - based on the entity, we should know the array shape.

Reply:
I agree.
But I thought this could have been a good first step.

list<array> is better than mixed,
then list<array{id: int, ...}> will be better than list<array>.

Also I had no idea how to improve the 
```
if ($isObject->yes()) {
    return new ArrayType(new MixedType(), new MixedType());
}
```
part with the PHPStan kit.